python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -150,14 +154,59 @@ export default function EditorContainer({
     [dispatch]
   );
 
+  const debouncedUpdateModel = useMemo(
+    () =>
+      debounce(
+        (promptName?: string, modelMetadata?: string | ModelMetadata) =>
+          callbacks.updateModel(promptName, modelMetadata),
+        250
+      ),
+    [callbacks.updatePrompt]
+  );
+
   const onUpdatePromptModelSettings = useCallback(
     async (promptIndex: number, newModelSettings: any) => {
       dispatch({
         type: "UPDATE_PROMPT_MODEL_SETTINGS",
         index: promptIndex,
         modelSettings: newModelSettings,
       });
-      // TODO: Call server-side endpoint to update model settings
+      // TODO: Call server-side endpoint to update model
+    },
+    [dispatch]
+  );
+
+  const onUpdatePromptModel = useCallback(
+    async (promptIndex: number, newModel?: string) => {
+      dispatch({
+        type: "UPDATE_PROMPT_MODEL",
+        index: promptIndex,
+        modelName: newModel,
+      });
+
+      try {
+        const prompt = clientPromptToAIConfigPrompt(
+          aiconfigState.prompts[promptIndex]
+        );
+        const currentModel = prompt.metadata?.model;
+        let modelData: string | ModelMetadata | undefined = newModel;
+        if (newModel && currentModel && typeof currentModel !== "string") {
+          modelData = {
+            ...currentModel,
+            name: newModel,
+          };
+        }
+
+        await debouncedUpdateModel(prompt.name, modelData);
+
+        // TODO: Consolidate
+      } catch (err: any) {
+        showNotification({
+          title: "Error updating prompt model",
```

Comment:
Would it be possible to read the http message that we return and we display it? Can be done future PR